### ISSUE TITLE
Inconsistent Panic Warning Detection for Similar Code

### ISSUE THREAD

**Issue by stoneman258** [lizhuohua/rust-mir-checker#16]
## Description
I encountered an inconsistency in `mirchecker` when testing two similar code cases. The first case correctly triggers a `panic` warning, but the second case does not, even though both cases should trigger a warning.

### Case 1
```rust
fn next(data: &[u8], ptr: *const u8) -> Option<&u8> {
    unsafe {
        debug_assert!(data.as_ptr() as usize <= ptr as usize);    //report warning
        Some(&*ptr)
    }
}
```
### Case 2
```rust
fn _next(data: &[u8], ptr: *const u8) -> Option<&u8> {
    unsafe {
        let next_closure = |data: usize, ptr: *const u8| {
            debug_assert!(data <= ptr as usize);    //should report warning
            Some(&*ptr)
        };
        next_closure(data.as_ptr() as usize, ptr)
    }
}
```

## Expected Behavior
Both cases should trigger a warning due to the `debug_assert` macro.

## Actual Behavior
Case 1: A warning is correctly detected.
Case 2: No warning is detected.
